### DISCUSSION TITLE
How do I actively reset the traceId in code so that the downstream node is another traceId.

### DISCUSSION THREAD

**Discussion by michaelwangwangwang** [apache/skywalking#10720] [Q&A] [answered]
There is an infinite loop in my code, and the downstream interface is called in the loop. I want to distinguish each call in this loop, so that each call generates a new traceId

**Comment by michaelwangwangwang** [comment]
please!!!

**Comment by wu-sheng** [comment, marked as answer]
No, you can't. This tracing system is following how the code works. This is not for tracing business processes.

**Reply by michaelwangwangwang** [reply]
I wrote an infinite loop, in which there are loop calls to downstream services, how do I distinguish between each call?

**Reply by wu-sheng** [reply]
Then, why do you start the trace before the loop? That seems your custom tracing.

**Reply by michaelwangwangwang** [reply]
I have a java microservice A. In addition to the common interface, this microservice also has a while (true) infinite loop logic. This infinite loop contains the logic of calling the downstream microservice B. I give this microservice A a Created a skywalking-agent for call chain tracing. Now it is found that the traceId of A calling B is the same, which is very difficult to troubleshoot.

**Reply by wu-sheng** [reply]
I don't think this is a typical case we would support. If there is HTTP inbound, an endless loop would cause a timeout. This should not happen.
How is this possible to be on the product?

**Reply by michaelwangwangwang** [reply]
There is indeed such a scenario. I once tried to modify the sw8 request header when calling microservice b, so as to modify the traceId during cross-process access, but found it useless. It seems that skywalking-agent will overwrite it. I may be in Do you get the skywalking context and reset the traceId in the code?

**Reply by michaelwangwangwang** [reply]
> I don't think this is a typical case we would support. If there is HTTP inbound, an endless loop would cause a timeout. This should not happen. How is this possible to be on the product?

This while(true) is not calling microservice b all the time, but is similar to a listener, which is called only when there is an external trigger.

**Reply by wu-sheng** [reply]
Then remove the plugin for the HTTP server.

**Reply by michaelwangwangwang** [reply]
If I delete it, can other interface calls of microservice A be connected?

**Reply by wu-sheng** [reply]
Of course. You can't have both, I am afraid. 
Like I mentioned, this is a reality of your service running, not a business process tracing.

**Reply by michaelwangwangwang** [reply]
I don’t know if skywalking can provide a method. The business can reset/clear the traceId stored in the skywalking context in the code, so that the link behind it will not be connected with the front, but a new start, using skywalking for the business a certain amount of flexibility.

**Reply by wu-sheng** [reply]
I repeated many times, NO, we are not.
Business tracing is not SkyWalking business.
